server/game/cards/08.5-TFM/SacrifieAuDieuRouge.js

```diff
@@ -0,0 +1,39 @@
+const DrawCard = require('../../drawcard.js');
+
+class SacrifieAuDieuRouge extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Search for a character',
+            phase: 'marshal',
+            cost: ability.costs.sacrifice(card => card.getType() === 'character'),
+            condition: () => this.controller.anyCardsInPlay(card => card.hasTrait('R\'hllor') && card.getType() === 'character'),
+            handler: context => {
+                this.game.promptForDeckSearch(this.controller, {
+                    activePromptTitle: 'Select a card',
+                    cardCondition: card =>
+                        card.getType() === 'character' &&
+                        card.hasTrait('R\'hllor') &&
+                        card.getCost() <= context.costs.sacrifice.getCost(),
+                    onSelect: (player, card) => this.cardSelected(player, card),
+                    onCancel: player => this.doneSelecting(player),
+                    source: this
+                });
+            }
+        });
+    }
+
+    cardSelected(player, card) {
+        player.moveCard(card, 'hand');
+        this.game.addMessage('{0} uses {1} to search their deck and add {2} to their hand',
```

Comment:
Very minor, but these messages would usually be phrased as '{Player} plays {event} and sacrifices {card} to...' (same for the message in the cancel method). Looks good otherwise.

Reply:
Thanks, you're right, fixed.
(Looks like we have a bunch of other deck search events that uses the bad wording, probably inherited from non-event tutoring cards…)